x/recovery/keeper/ibc_callbacks.go

```diff
@@ -64,54 +65,49 @@ func (k Keeper) OnRecvPacket(
 		)
 	}
 
-	// case 1: sender ≠ recipient.
-	// Recovery is only possible for addresses in which the sender = recipient
-	// (i.e transferring to your own account in Evmos).
+	// Check if sender != recipient, as recovery is only possible for transfers to
+	// a sender's own account on Evmos (sender == recipient)
 	if !sender.Equals(recipient) {
 		// Continue to the next IBC middleware by returning the original ACK.
 		return ack
 	}
 
-	// get the sender account
-	account := k.accountKeeper.GetAccount(ctx, sender)
+	// get the recipient/sender account
+	account := k.accountKeeper.GetAccount(ctx, recipient)
 
 	// recovery is not supported for vesting or module accounts
-	_, isVestingAcc := account.(vestexported.VestingAccount)
-	if isVestingAcc {
+	if _, isVestingAcc := account.(vestexported.VestingAccount); isVestingAcc {
 		return ack
 	}
 
-	_, isModuleAccount := account.(authtypes.ModuleAccountI)
-	if isModuleAccount {
+	if _, isModuleAccount := account.(authtypes.ModuleAccountI); isModuleAccount {
 		return ack
 	}
 
-	// Case 2. sender pubkey is a supported key (eth_secp256k1, amino multisig, ed25519)
-	// ==> Continue and return success ACK as the funds are not stuck on chain
+	// Check if recipient pubkey is a supported key (eth_secp256k1, amino multisig,
+	// ed25519). Continue and return success ACK as the funds are not stuck on
+	// chain for supported keys
 	if account != nil && evmos.IsSupportedKey(account.GetPubKey()) {
 		return ack
 	}
 
-	// NOTE: Since destination channel is authorized and not from an EVM chain, we know that
-	// only secp256k1 keys are supported in the source chain. This means that we can now
-	// initiate the recovery logic
-
-	// transfer the balance back to the sender address
-	destPort := packet.DestinationPort
-	destChannel := packet.DestinationChannel
+	// Perform recovery to transfer the balance back to the sender bech32 address.
+	// NOTE: Since destination channel is authorized and not from an EVM chain, we
+	// know that only secp256k1 keys are supported in the source chain.
 	balances := sdk.Coins{}
 
-	// iterate over all the tokens owned by the address (i.e sender balance) and
+	// iterate over all tokens owned by the address (i.e sender balance) and
 	// transfer them to the original sender address in the source chain (if
 	// applicable, see cases for IBC vouchers below).
-	k.bankKeeper.IterateAccountBalances(ctx, sender, func(coin sdk.Coin) (stop bool) {
+	k.bankKeeper.IterateAccountBalances(ctx, recipient, func(coin sdk.Coin) (stop bool) {
 		if coin.IsZero() {
 			// safety check: continue
 			return false
 		}
 
 		if strings.HasPrefix(coin.Denom, "ibc/") {
-			// IBC vouchers, obtain the source port and channel from the denom path
+			var destPort, destChannel string
```

Comment:
these should be declared on top of the iterator to avoid allocating memory on every item